lib/quicly.c

```diff
@@ -1994,6 +2068,9 @@ static int do_detect_loss(quicly_loss_t *ld, int64_t now, uint64_t largest_acked
         quicly_acks_release(&conn->egress.acks, &iter);
         quicly_acks_next(&iter);
     }
+    conn->egress.cc.bytes_in_flight -= conn->egress.cc.this_ack.nbytes;
```

Comment:
You should guard this. A massive bytes_in_flight will basically block the connection competely.
conn->egress.cc.bytes_in_flight -= (conn->egress.cc.bytes_in_flight < conn->egress.cc.this_ack.nbytes) ?
     conn->egress.cc.bytes_in_flight : conn->egress.cc.this_ack.nbytes;